src/irr/asset/COBJMeshFileLoader.cpp

```diff
@@ -300,51 +295,36 @@ asset::SAssetBundle COBJMeshFileLoader::loadAsset(io::IReadFile* _file, const as
 				else
 				{
 					v.normal32bit = 0;
-					currMtl->RecalculateNormals=true;
+                    recalcNormals.back() = true;
 				}
 
-				int vertLocation;
-				core::map<SObjVertex, int>::iterator n = currMtl->VertMap.find(v);
-				if (n!=currMtl->VertMap.end())
-				{
-					vertLocation = n->second;
-				}
+				uint32_t ix;
+				auto vtx_ix = map_vtx2ix.find(v);
+				if (vtx_ix != map_vtx2ix.end())
+					ix = vtx_ix->second;
 				else
 				{
-					currMtl->Vertices.push_back(v);
-					vertLocation = currMtl->Vertices.size() -1;
-					currMtl->VertMap.insert(std::pair<SObjVertex, int>(v, vertLocation));
+					ix = vertices.size();
+					vertices.push_back(v);
+					map_vtx2ix.insert({v, ix});
 				}
 
-				faceCorners.push_back(vertLocation);
+				faceCorners.push_back(ix);
 
 				// go to next vertex
 				linePtr = goNextWord(linePtr, endPtr);
 			}
 
-			// triangulate the face
-			for ( uint32_t i = 1; i < faceCorners.size() - 1; ++i )
-			{
-				// Add a triangle
-				performActionBasedOnOrientationSystem
-				(
-					[&]()
-					{
-						currMtl->Indices.push_back(faceCorners[0]);
-						currMtl->Indices.push_back(faceCorners[i]);
-						currMtl->Indices.push_back(faceCorners[i + 1]);
-					}, 
-				
-					[&]() 
-					{
-						currMtl->Indices.push_back(faceCorners[i + 1]);
-						currMtl->Indices.push_back(faceCorners[i]);
-						currMtl->Indices.push_back(faceCorners[0]);
-					}
-				);
-			}
-			faceCorners.resize(0); // fast clear
-			faceCorners.reserve(32);
```

Comment:
OBJ can spec faces which are quads or even polygons, we need triangulation if it used to work before.

Reply:
nah, it always assumed triangles. I dont why there was reserve(32) then so i changed it to 3

Reply:
XDD obviously you were right, there was triangulation

Reply:
how did you fix?

Reply:
Wasn't it like that at beginning?

Reply:
ok I see the new change now.